coordinator/src/orderbook/trading.rs

```diff
@@ -0,0 +1,191 @@
+use crate::orderbook::routes::MatchParams;
+use crate::orderbook::routes::Order;
+use crate::orderbook::routes::OrderType;
+use anyhow::Result;
+use rust_decimal::Decimal;
+use std::cmp::Ordering;
+use trade::Direction;
+
+/// Matches a provided market order with limit orders from the DB
+///
+/// If the order is a long order, we return the orders with the highest price
+/// If the order is a short order, we return the orders with the lowest price
+pub fn match_order(order: Order, all_orders: Vec<Order>) -> Result<Vec<MatchParams>> {
```

Comment:
🤔 I am a bit confused - we are not matching the order by `Direction` as far as I can see. 

A long order can only be matched with a short order of the other party!
As far as I can see there is no filter step, but we match against all order.
If we would have only 2 long orders we would actually match them with each other...? Am I missing something? 😅

Reply:
The list of possible orders is passed in and already filtered outside.

Reply:
>  If we would have only 2 long orders (one market, one limit) we would actually match them with each other...? Am I missing something? 😅

No, if we have 2 long orders they would not match with each other :)

Reply:
OK, I followed it back to `all_by_direction_and_type` :) - adding a few comments on how we could potentially improve the readability.

Reply:
[`a9cd5d9` (#248)](https://github.com/get10101/10101/pull/248/commits/a9cd5d9126caf2c5ac3e8effa5c118f4fbd27f2d)